### PR TITLE
feat: Support multiple vector indexes in a collection

### PR DESCRIPTION
**Pull request by xige-16**
issue: #25639 

/kind improvement
Signed-off-by: xige-16 <xi.ge@zilliz.com>

### CODE REVIEW

File: internal/datacoord/util.go [hunk 1]
```diff
@@ -189,3 +190,20 @@ func parseBuildIDFromFilePath(key string) (UniqueID, error) {
 	}
 	return strconv.ParseInt(ss[len(ss)-1], 10, 64)
 }
+
+func getFieldByID(handler Handler, collectionID UniqueID, fieldID int64) (*schemapb.FieldSchema, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*2)
+	coll, err := handler.GetCollection(ctx, collectionID)
+	cancel()
```
Comment (jiaoew1991) [inline internal/datacoord/util.go:197]: defer cancel() ?

File: internal/datacoord/util.go [hunk 2]
```diff
@@ -189,3 +190,20 @@ func parseBuildIDFromFilePath(key string) (UniqueID, error) {
...
+	if err != nil {
+		return nil, fmt.Errorf("failed to get collection %d", collectionID)
```
Comment (jiaoew1991) [inline internal/datacoord/util.go:199]: should use merr

File: internal/proxy/util.go [hunk 3]
```diff
@@ -1505,6 +1505,30 @@ func checkDynamicFieldData(schema *schemapb.CollectionSchema, insertMsg *msgstre
 	return nil
 }
 
+func getIndexedField(ctx context.Context, dbName, collectionName, fieldName string) (*schemapb.FieldSchema, error) {
```
Comment (xiaocai2333) [inline internal/proxy/util.go:1508]: rename to getField?

**Comment by xige-16** [comment]
rerun ut

**Pull request merged**